src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -326,60 +326,56 @@ internal async ValueTask<SemanticModel> GetRequiredNullableDisabledSemanticModel
     /// </returns>
     private async Task<SemanticModel?> GetSemanticModelHelperAsync(bool disableNullableAnalysis, CancellationToken cancellationToken)
     {
-        try
-        {
-            if (!this.SupportsSemanticModel)
-            {
-                return null;
-            }
+        if (!this.SupportsSemanticModel)
+            return null;
 
-            SemanticModel? semanticModel;
-            if (disableNullableAnalysis)
+        var semanticModel = await GetSemanticModelWorkerAsync().ConfigureAwait(false);
+        this.Project.Solution.OnSemanticModelObtained(this.Id, semanticModel);
```

Comment:
>OnSemanticModelObtained [](http://example.com/codeflow?start=30&length=23)

No need to indicate in the cache the value of disableNullableAnalysis?

Reply:
Never mind, it's never queried from that cache, strictly used as a mechanism to keep it alive.

Reply:
(although it looks like this won't be able to keep alive both the semantic models for the active document)

Reply:
Finally looked in the SemanticModelCaching code, and it's able to detect that and keeps both. Nice!

Reply:
Haha. A wild journey in four steps. :-)